src/Rings/mpoly.jl

```diff
@@ -161,7 +161,16 @@ using .Orderings
 #type for orderings, use this...
 #in general: all algos here needs revision: do they benefit from gb or not?
 
-default_ordering(R::MPolyRing) = degrevlex(R)
+function default_ordering(R::MPolyRing)
+  if has_attribute(R, :default_ordering)
+    return get_attribute(R, :default_ordering)
+  else
+    default_ordering = degrevlex(R)
+    set_attribute!(R, :default_ordering => default_ordering)
+    return default_ordering
+  end
+end
+
```

Comment:
```suggestion
function default_ordering(R::MPolyRing)
  return get_attribute!(R, :default_ordering) do
    degrevlex(R)
  end
end
```

I would suggest to simplify it as above, as it is more readable and should be a hinge faster, as it avoids some attribute lookup. See https://nemocas.github.io/AbstractAlgebra.jl/dev/misc/#AbstractAlgebra.get_attribute!